patches/server/0764-Added-ability-to-control-player-s-insomnia-and-phant.patch

```diff
@@ -0,0 +1,143 @@
++            int temp = phantomsSpawnAttemptMinSeconds;
++            phantomsSpawnAttemptMinSeconds = phantomsSpawnAttemptMaxSeconds;
++            phantomsSpawnAttemptMaxSeconds = temp;
++        }
++    }
++
+     public static String authenticationServersDownKickMessage = ""; // empty = use translatable message
+     private static void authenticationServersDownKickMessage() {
+         authenticationServersDownKickMessage = Strings.emptyToNull(getString("messages.kick.authentication-servers-down", authenticationServersDownKickMessage));
+diff --git a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+index 261633687b5dd55559b2ddb305c3a21a8b308ada..e9fc868c24b44d2906f68659136114114db6bf32 100644
+--- a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
++++ b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+@@ -710,6 +710,11 @@ public class PaperWorldConfig {
+         phantomOnlyAttackInsomniacs = getBoolean("phantoms-only-attack-insomniacs", phantomOnlyAttackInsomniacs);
+     }
+ 
++    public int playerInsomniaStartTicks = 72000;
++    private void playerInsomniaStartTicks() {
++        playerInsomniaStartTicks = getInt("player-insomnia-start-ticks", playerInsomniaStartTicks);
++    }
++
+     public int noTickViewDistance;
+     private void viewDistance() {
+         this.noTickViewDistance = this.getInt("viewdistances.no-tick-view-distance", -1);
+diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
+index 25c3a96ca88494650d93efcaeecf5ae08ed0810f..3dc2360631ec0168b0200d152f7cf6a396a0943a 100644
+--- a/src/main/java/net/minecraft/server/MinecraftServer.java
++++ b/src/main/java/net/minecraft/server/MinecraftServer.java
+@@ -569,7 +569,17 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
+             boolean flag = generatorsettings.isDebug();
+             long i = generatorsettings.seed();
+             long j = BiomeManager.obfuscateSeed(i);
+-            List<CustomSpawner> list = ImmutableList.of(new PhantomSpawner(), new PatrolSpawner(), new CatSpawner(), new VillageSiege(), new WanderingTraderSpawner(iworlddataserver));
++            // Paper start
++            List<CustomSpawner> list = Lists.newArrayList(new PatrolSpawner(), new CatSpawner(), new VillageSiege(), new WanderingTraderSpawner(worlddata));
++
++            int phantomsSpawnAttemptMinSeconds = com.destroystokyo.paper.PaperConfig.phantomsSpawnAttemptMinSeconds;
++            int phantomsSpawnAttemptMaxSeconds = com.destroystokyo.paper.PaperConfig.phantomsSpawnAttemptMaxSeconds;
++            if ((phantomsSpawnAttemptMinSeconds >= 0 && phantomsSpawnAttemptMaxSeconds >= 0) && (phantomsSpawnAttemptMinSeconds != 0 || phantomsSpawnAttemptMaxSeconds != 0)) {
++                list.add(new PhantomSpawner());
++            }
```

Comment:
I would move the logic that guards the new PhantomSpawner to a method inside Paper config since its duplicated in the CraftServer#createWorld.